app/addons/documents/shared-views.js

```diff
@@ -160,12 +160,20 @@ function(app, FauxtonAPI, Components, Documents, Databases) {
     className:  "nav nav-list",
     template: "addons/documents/templates/design_doc_menu",
     events: {
-      "click .js-collapse-toggle": "toggleArrow"
+      "click .js-collapse-toggle": "toggleArrow",
+      "click .js-toggle-submenu": 'toggleSubmenu'
     },
 
     toggleArrow:  function(e){
       this.$(e.currentTarget).toggleClass("down");
+      $('#' + this.model.id.replace(/^_design\//,"").replace(/\./g, '\\.')).collapse('toggle'); 
+      //if there is a '.' in the design doc name, jquery interprets that as a class
+      //see assets/js/libs/bootstrap.js line 1739
     },
+    toggleSubmenu: function (e) {
+      this.$(e.currentTarget).toggleClass("down");
+    },
+
```

Comment:
please follow the codingguidelines:

newline between properties,
single quotes,
space after comma